bitcoin/src/consensus/encode.rs

```diff
@@ -386,100 +456,22 @@ impl_int_encodable!(i16, read_i16, emit_i16);
 impl_int_encodable!(i32, read_i32, emit_i32);
 impl_int_encodable!(i64, read_i64, emit_i64);
 
-#[allow(clippy::len_without_is_empty)] // VarInt has on concept of 'is_empty'.
-impl VarInt {
-    /// Gets the length of this VarInt when encoded.
-    #[inline]
-    #[deprecated(since = "0.31.0", note = "use size instead")]
-    pub const fn len(&self) -> usize { self.size() }
-
-    /// Returns the number of bytes this varint contributes to a transaction size.
-    ///
-    /// Returns 1 for 0..=0xFC, 3 for 0xFD..=(2^16-1), 5 for 0x10000..=(2^32-1), and 9 otherwise.
-    #[inline]
-    pub const fn size(&self) -> usize {
-        match self.0 {
-            0..=0xFC => 1,
-            0xFD..=0xFFFF => 3,
-            0x10000..=0xFFFFFFFF => 5,
-            _ => 9,
-        }
-    }
-}
-
-/// Implements `From<T> for VarInt`.
+/// Returns the number of bytes needed for the varint encoding of this integer.
 ///
-/// `VarInt`s are consensus encoded as `u64`s so we store them as such. Casting from any integer size smaller than or equal to `u64` is always safe and the cast value is correctly handled by `consensus_encode`.
-macro_rules! impl_var_int_from {
-    ($($ty:tt),*) => {
-        $(
-            /// Creates a `VarInt` from a `usize` by casting the to a `u64`.
-            impl From<$ty> for VarInt {
-                fn from(x: $ty) -> Self { VarInt(x as u64) }
-            }
-        )*
-    }
-}
-impl_var_int_from!(u8, u16, u32, u64, usize);
-
-impl Encodable for VarInt {
-    #[inline]
-    fn consensus_encode<W: io::Write + ?Sized>(&self, w: &mut W) -> Result<usize, io::Error> {
-        match self.0 {
-            0..=0xFC => {
-                (self.0 as u8).consensus_encode(w)?;
-                Ok(1)
-            }
-            0xFD..=0xFFFF => {
-                w.emit_u8(0xFD)?;
-                (self.0 as u16).consensus_encode(w)?;
-                Ok(3)
-            }
-            0x10000..=0xFFFFFFFF => {
-                w.emit_u8(0xFE)?;
-                (self.0 as u32).consensus_encode(w)?;
-                Ok(5)
-            }
-            _ => {
-                w.emit_u8(0xFF)?;
-                self.0.consensus_encode(w)?;
-                Ok(9)
-            }
-        }
-    }
+/// Returns 1 for 0..=0xFC, 3 for 0xFD..=(2^16-1), 5 for 0x10000..=(2^32-1), and 9 otherwise.
```

Comment:
Is there like a BIP or something that can be referenced here?

Reply:
Nah, this is all super old stuff. Might be somewhere on wiki.bitcoin.it, but mostly just Core's ref impl.